From 444d27bbcc7017061a2a49c5dbc5865d34fc7a65 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Tue, 7 May 2019 10:06:04 +0100
Subject: [PATCH 501/773] overlays: Remove upstream-aux-interrupt overlay

We no longer have a downstream-specific auxilliary interrupt
driver, so the overlay to disable it is no longer needed.

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 arch/arm/boot/dts/overlays/Makefile           |  1 -
 arch/arm/boot/dts/overlays/README             | 12 +++----
 .../upstream-aux-interrupt-overlay.dts        | 33 -------------------
 .../boot/dts/overlays/upstream-overlay.dts    |  2 +-
 4 files changed, 6 insertions(+), 42 deletions(-)
 delete mode 100644 arch/arm/boot/dts/overlays/upstream-aux-interrupt-overlay.dts

--- a/arch/arm/boot/dts/overlays/Makefile
+++ b/arch/arm/boot/dts/overlays/Makefile
@@ -151,7 +151,6 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	uart1.dtbo \
 	udrc.dtbo \
 	upstream.dtbo \
-	upstream-aux-interrupt.dtbo \
 	vc4-fkms-v3d.dtbo \
 	vc4-kms-kippah-7inch.dtbo \
 	vc4-kms-v3d.dtbo \
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -2206,18 +2206,16 @@ Params: alsaname                Name of
 
 
 Name:   upstream
-Info:   Allow usage of downstream .dtb with upstream kernel. Comprises
-        vc4-kms-v3d, dwc2 and upstream-aux-interrupt overlays.
+Info:   Allow usage of downstream .dtb with upstream kernel. Comprises the
+        vc4-kms-v3d and dwc2 overlays.
 Load:   dtoverlay=upstream
 Params: <None>
 
 
 Name:   upstream-aux-interrupt
-Info:   Allow usage of downstream .dtb with upstream kernel by binding AUX
-        devices directly to the shared AUX interrupt line. One of the parts
-        of the 'upstream' overlay
-Load:   dtoverlay=upstream-aux-interrupt
-Params: <None>
+Info:   This overlay has been deprecated and removed because it is no longer
+        necessary.
+Load:   <Deprecated>
 
 
 Name:   vc4-fkms-v3d
--- a/arch/arm/boot/dts/overlays/upstream-aux-interrupt-overlay.dts
+++ /dev/null
@@ -1,33 +0,0 @@
-// Overlay for missing AUX interrupt controller
-// Instead we bind all AUX devices to the generic AUX interrupt line
-/dts-v1/;
-/plugin/;
-
-/ {
-    compatible = "brcm,bcm2708";
-
-    fragment@0 {
-        target = <&uart1>;
-        __overlay__ {
-            interrupt-parent = <&intc>;
-            interrupts = <0x1 0x1d>;
-        };
-    };
-
-    fragment@1 {
-        target = <&spi1>;
-        __overlay__ {
-            interrupt-parent = <&intc>;
-            interrupts = <0x1 0x1d>;
-        };
-    };
-
-    fragment@2 {
-        target = <&spi2>;
-        __overlay__ {
-            interrupt-parent = <&intc>;
-            interrupts = <0x1 0x1d>;
-        };
-    };
-};
-
--- a/arch/arm/boot/dts/overlays/upstream-overlay.dts
+++ b/arch/arm/boot/dts/overlays/upstream-overlay.dts
@@ -1,4 +1,4 @@
-// redo: ovmerge -c vc4-kms-v3d-overlay.dts,cma-96 dwc2-overlay.dts,dr_mode=otg upstream-aux-interrupt-overlay.dts,
+// redo: ovmerge -c vc4-kms-v3d-overlay.dts,cma-96 dwc2-overlay.dts,dr_mode=otg
 
 /dts-v1/;
 /plugin/;
